File: schemas/social-media/convex/follows.ts
// follows: follower-following relationships between users with notification preferences.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const follows = defineTable({
  followerId: v.id("users"),
  followingId: v.id("users"),
  notify: v.boolean(),
  showReposts: v.boolean(),
})
  .index("by_follower_id_following_id", ["followerId", "followingId"])
  .index("by_following_id", ["followingId"]);

File: schemas/social-media/spacetimedb/follows.rs
// follows: follower-following relationships between users.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = follows, public)]
pub struct Follow {
    #[primary_key]
    pub id: String, // UUID
    // Composite unique: (follower_id, following_id) — enforce in reducer logic.
    #[index(btree)]
    pub follower_id: String, // UUID — FK → users.id (cascade delete)
    #[index(btree)]
    pub following_id: String, // UUID — FK → users.id (cascade delete)
    pub notify: bool,
    pub show_reposts: bool,
    pub created_at: Timestamp,
}
